src/avram/model.cr

```diff
@@ -184,6 +184,79 @@ class Avram::Model
     {% COLUMNS << {name: type_declaration.var, type: data_type, nilable: nilable.id, autogenerated: autogenerated} %}
   end
 
+  macro polymorphic_belongs_to(type_declaration, autogenerated = false)
+
+    {% nilable = false %}
+    {% for type in type_declaration.type.types %}
+      {% if type.class_name == "Path" %}
+        {% nilable = true if type.names[0] == "Nil" %}
+      {% elsif type.class_name == "Union" %}
+        {% nilable = true if type.types.any? { |item| item.names[0] == "Nil" } %}
+      {% end %}
+    {% end %}
+
+    {% COLUMNS << {name: type_declaration.var + "_id", type: Int64, nilable: nilable.id, autogenerated: autogenerated} %}
+    {% COLUMNS << {name: type_declaration.var + "_type", type: String, nilable: nilable.id, autogenerated: autogenerated} %}
+
+    macro finished
+      class SaveOperation
+        before_save :validates_polymorphic_association
+
+        private def validates_polymorphic_association
+          previous_def
```

Comment:
Instead of naming this and using `previous_def` you can now do `before_save` with a block:

```crystal
before_save do
  # the code
end
```

Reply:
awesome!!! 🎉